src/goto-symex/symex_assign.cpp

```diff
@@ -228,7 +228,8 @@ void symex_assignt::assign_non_struct_symbol(
   state.record_events.pop();
 
   auto current_assignment_type =
-    ns.lookup(l2_lhs.get_object_name()).is_auxiliary
+    ns.lookup(l2_lhs.get_object_name()).is_auxiliary &&
+        id2string(l2_lhs.get_object_name()).find("__SM") != std::string::npos
```

Comment:
💡 There should probably be a define for this constant.

Reply:
Good idea.

Changed